R/yearquarter.R

```diff
@@ -282,6 +282,8 @@ format.yearquarter <- function(x, format = "%Y Q%q", ...) {
   x <- as_date(x)
   qtr <- quarter(x)
   qtr_sub <- map_chr(qtr, function(z) gsub("%q", z, x = format))
+  qtr_sub[is.na(qtr_sub)] <- "-" # NA formats cause errors
```

Comment:
If `format = "%Y-Q%q"`, this probably won't work.
```suggestion
  qtr_sub[is.na(qtr_sub)] <- "" # NA formats cause errors
```

Reply:
Works fine. The reason I choose `"-"` not `""` is because `base:::format.POSIXlt` has special computation when `format = ""`. We don't need that (unless the user specifies `format = ""` for non-NA entries), as we just want to return NA for NA cases.

Reply:
``` r
library(tsibble)
yq <- vctrs::vec_c(NA, yearquarter("1970 Q1"))
format(yq, format = "%Y-Q%q")
#> [1] NA        "1970-Q1"
```

<sup>Created on 2020-03-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Okay. Please add a unit test.